internal/pkg/encryption/keys/generated.go

```diff
@@ -0,0 +1,46 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package keys
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/talos-systems/go-smbios/smbios"
+)
+
+// GeneratedKeyHandler generates the key based on current node information
+// and provided template string.
+type GeneratedKeyHandler struct {
+}
+
+// NewGeneratedKeyHandler creates new GeneratedKeyHandler.
+func NewGeneratedKeyHandler() (*GeneratedKeyHandler, error) {
+	return &GeneratedKeyHandler{}, nil
+}
+
+// GetKey implements KeyHandler interface.
+func (h *GeneratedKeyHandler) GetKey(options ...KeyOption) ([]byte, error) {
+	opts, err := NewDefaultOptions(options)
+	if err != nil {
+		return nil, err
+	}
+
+	s, err := smbios.New()
+	if err != nil {
+		return nil, err
+	}
+
+	machineUUID, err := s.SystemInformation().UUID()
+	if err != nil {
+		return nil, err
+	}
+
+	if machineUUID == uuid.Nil {
```

Comment:
we need to check for randomness of UUID here - could be a follow-up ticket, but we need to have it in 0.9

Reply:
Yeah, but I need help with forming randomness criteria/threshold.
We can treat uuid not random if it consists of 0 more than some ratio I guess, but I'm just not sure what level of check will be enough.
I was trying to google any libraries or methods which do that, but I couldn't find anything. Maybe I was just searching for the wrong keywords.